BlueprintUICommonControls/Sources/AttributedLabel.swift

```diff
@@ -96,27 +96,46 @@ public struct AttributedLabel: Element, Hashable {
         configure(&self)
     }
 
-    public var content: ElementContent {
-        struct Measurer {
-            private static let prototypeLabel = LabelView()
+    // MARK: Element
 
-            func measure(model: AttributedLabel, in constraint: SizeConstraint, environment: Environment) -> CGSize {
-                let label = Self.prototypeLabel
-                label.update(model: model, environment: environment, isMeasuring: true)
-                return label.sizeThatFits(constraint.maximum)
-            }
+    /// The text to pass to the underlying `UILabel`, normalized for display if necessary.
+    var displayableAttributedText: NSAttributedString {
+        if needsTextNormalization || linkDetectionTypes.isEmpty == false {
+            return attributedText.normalizingForView(with: numberOfLines)
+        } else {
+            return attributedText
         }
+    }
+
+    @_spi(BlueprintAttributedLabel)
+    /// Set this if you can guarantee that your label implementation will not need string normalization.
+    /// For example, a custom label type wrapping `AttributedLabel` may set this to `false`.
+    /// You can check if this value should be false via `NSAttributedString.needsNormalizingForView(...)`
+    public var needsTextNormalization: Bool = true
+
+    private static let prototypeLabel = LabelView()
+
+    public var content: ElementContent {
 
-        return ElementContent { sizeConstraint, environment -> CGSize in
-            Measurer().measure(model: self, in: sizeConstraint, environment: environment)
+        // We create this outside of the measurement block so it's called fewer times.
+        let text = displayableAttributedText
+
+        return ElementContent { constraint, environment -> CGSize in
+            let label = Self.prototypeLabel
+            label.update(model: self, text: text, environment: environment, isMeasuring: true)
+            return label.sizeThatFits(constraint.maximum)
```

Comment:
Is the goal of this change to avoid making the instance of `Measurer`, or to simplify things by removing a type?

If we like the encapsulation of `Measurer`, we could update `measuer(model:in:environment:)` to a static method.

Reply:
Measurer just didn't do much; so I figured I'd remove it given we now have an `ElementContent` that gives us a callback to do the measurement.